### ISSUE TITLE
`InverseCancellation` can invalidly cancel parametric gates

### ISSUE THREAD

**Issue by jakelishman** [Qiskit/qiskit#11815]
### Environment

- **Qiskit version**: 1.0.0rc1
- **Python version**: 3.10
- **Operating system**: macOS

### What is happening?

The `InverseCancellation` pass can, in some circumstances, cancel two parametric gates that should not cancel, despite some of our handling and testing against this added in #11211

### How can we reproduce the issue?

```python
from math import pi

from qiskit.circuit.library import RZGate
from qiskit.circuit import QuantumCircuit
from qiskit.transpiler.passes import InverseCancellation

inverse_pass = InverseCancellation([RZGate(0)])

qc = QuantumCircuit(1)
qc.rz(0, 0)
qc.rz(pi, 0)

inverse_pass(qc)
```
returns a circuit with the gates cancelled.

### What should happen?

Those two gates shouldn't match any of the given rules, so shouldn't have been cancelled.

**Comment by sesajad** [comment]
Hi,
It seems to be due to the fact that the following condition is not checked for the last gate (last gate in a run).

https://github.com/Qiskit/qiskit/blob/5ae4790cf6f7ca918b2d1cb9a740bd4287171c30/qiskit/transpiler/passes/optimization/inverse_cancellation.py#L115

Therefore, it can be reproduced by replacing `qc.rz(pi, 0)` with any other gate with the same name.

I have sent a PR to fix it.

**Comment by jakelishman** [comment]
Fixed by #11834
